jax/interpreters/masking.py

```diff
@@ -432,6 +452,46 @@ def todo(vals):
       return map(partial(MaskTracer, trace), vals, shapes)
     return vals, todo
 
+  def process_map(self, map_primitive, f, tracers, params):
+    vals, shapes = unzip2((t.val, t.polymorphic_shape) for t in tracers)
+    if not any(is_polymorphic(s) for s in shapes):
+      return map_primitive.bind(f, *vals, **params)
+    mapped_invars = params['mapped_invars']
+    logical_env, padded_env = shape_envs
+    poly_axis_size, = {shape[0]
+                       for shape, mapped_invar in zip(shapes, mapped_invars)
+                       if mapped_invar}
+    padded_axis_size = eval_poly(poly_axis_size, padded_env)
+    env_keys, padded_env_vals = unzip2(sorted(padded_env.items()))
+    params = dict(
+        params,
+        mapped_invars=((False,) * len(logical_env) + params['mapped_invars']),
+        axis_size=padded_axis_size)
+    shapes = tuple(shape[1:] if mapped_invar else shape
```

Comment:
I think it's best to use `core.mapped_aval` here, in part because it handles units and in part because using that abstraction will help us revise things later.